module/finalizer/collection/finalizer.go

```diff
@@ -22,15 +21,15 @@ import (
 type Finalizer struct {
 	db           *badger.DB
 	transactions mempool.Transactions
-	prov         network.Engine
+	prov         collection.GuaranteedCollectionPublisher
```

Comment:
The Finalizer calls this a "provider", which is maybe a bit confusing/inconsistent with the rest of the code - should this field be renamed to "pusher" or "publisher" or something else?

Reply:
Agree - I think `pusher` is reasonable since that is the name of the relevant engine's package.

Reply:
Applied this change in https://github.com/onflow/flow-go/pull/6780/commits/eacc99232398a224a88445f20c6a4f6945046718.